OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs

```diff
@@ -121,11 +121,9 @@ public override void DoImpact(Target target, Actor firedBy, IEnumerable<int> dam
 			var explosion = Explosions.RandomOrDefault(world.LocalRandom);
 			if (Image != null && explosion != null)
 			{
-				if (ForceDisplayAtGroundLevel)
-				{
-					var dat = world.Map.DistanceAboveTerrain(pos);
+				var dat = world.Map.DistanceAboveTerrain(pos);
+				if (ForceDisplayAtGroundLevel || dat < WDist.Zero)
```

Comment:
> To avoid Z-sorting issues when triggered from places
that might put the position below ground level, clamp vertical position
of sprite effect to at least ground level.

IMO this seems bogus. If the mod intends for the warhead to detonate below ground level, then moving the explosion to ground level is wrong. If the mod didn't intend the warhead to detonate below ground level, then thats a bug that needs to be solved at its source.

It would be better to treat the impact as invalid and skip the effect.